### ISSUE TITLE
Debian installation doesn't work

### ISSUE THREAD

**Issue by olehdevua** [fluent/fluent-bit#5865]
## Bug Report

**Describe the bug**

After following [instructions from official site](https://docs.fluentbit.io/manual/installation/linux/debian),
I was not able to install `debian` package.
The problem is that `$CODENAME` variable in bashscript is determined in incorrect way, when
user has `/etc/os-release` file.

**To Reproduce**

```bash
curl https://raw.githubusercontent.com/fluent/fluent-bit/master/install.sh | sh
```

**Expected behavior**

Package should be installed.

**Screenshots**

![image](https://user-images.githubusercontent.com/4011317/183911755-3766dd7e-d91f-4931-8e49-9fa548ed6797.png)

**Your Environment**

OS: Linux debian 5.18.0-3-amd64 #1 SMP PREEMPT_DYNAMIC Debian 5.18.14-1 (2022-07-23) x86_64 GNU/Linux
Package manager: apt

**Additional context**

**Comment by patrick-stephens** [comment]
Note Debian Bookworm is not supported: https://docs.fluentbit.io/manual/installation/supported-platforms
It is not a stable or released version either: https://wiki.debian.org/DebianReleases

Please submit a PR to resolve the issue with the install script.
Does the manual installation approach work as documented, i.e. not the single-line install?

**Comment by bmarinov** [comment]
> Note Debian Bookworm is not supported: https://docs.fluentbit.io/manual/installation/supported-platforms It is not a stable or released version either: https://wiki.debian.org/DebianReleases
> 
> Please submit a PR to resolve the issue with the install script. Does the manual installation approach work as documented, i.e. not the single-line install?

Not OP but here are some details on the issue:
I started with the manual steps since im testing bookworm and hit this issue:
```
The following packages have unmet dependencies:
 fluent-bit : Depends: libssl1.1 (>= 1.1.1) but it is not installable
```

This sound like https://github.com/fluent/fluent-bit/issues/5374 but I thought that libssl3 compatibility is now backed into the package? This is an ARM machine by the way. Any way to manually patch this on my end? Cheers!

**Comment by patrick-stephens** [comment]
We don't have a package for Bookworm explicitly so I'm guessing you're using one of the other targets, Bullseye?
For building it we just use the OS provided dependencies, nothing fancy: https://github.com/fluent/fluent-bit/blob/2049e320eb9c6c48d34b6c5c0ea46ef80e242124/packaging/distros/debian/Dockerfile#L64

I will note that is explicitly selecting 1.1 library.

**Comment by bmarinov** [comment]
> We don't have a package for Bookworm explicitly so I'm guessing you're using one of the other targets, Bullseye? For building it we just use the OS provided dependencies, nothing fancy:
> 
> https://github.com/fluent/fluent-bit/blob/2049e320eb9c6c48d34b6c5c0ea46ef80e242124/packaging/distros/debian/Dockerfile#L64
> 
> I will note that is explicitly selecting 1.1 library.

Ah of course, you are right! I tried using the Bullseye package. And after a night of sleep I can see why this wouldn't work. So I guess the right way to approach this is to build from source for the time being.

**Comment by patrick-stephens** [comment]
@bmarinov it likely is just a case of extending that Dockerfile then you can just run the `./packaging/build.sh -d debian/bookworm` script as we do for the releases. I'd hope there is no major change required from that.

You could then submit a PR to provide it back.

**Comment by bmarinov** [comment]
> @bmarinov it likely is just a case of extending that Dockerfile then you can just run the `./packaging/build.sh -d debian/bookworm` script as we do for the releases. I'd hope there is no major change required from that.
> 
> You could then submit a PR to provide it back.

Dropped a quick PR @ https://github.com/fluent/fluent-bit/pull/6409 and adding details as I test.

**Comment by patrick-stephens** [comment]
Probably should clarify this won't be available until the next release.

**Comment by patrick-stephens** [comment]
2.0.6 is up for everything except Bookworm, tracking under https://github.com/fluent/fluent-bit/issues/6442 to just add the repo.

**Comment by patrick-stephens** [comment]
Debian Bookworm packages should now be available for versions 2.0.6+ 
It seems to work fine for me with the install script as well, at least when I test in containers.
```shell
$ docker run --rm -it debian:bookworm /bin/sh -c 'apt-get update && apt-get install -y sudo gpg curl && curl https://raw.githubusercontent.com/fluent/fluent-bit/master/install.sh | sh'
...
Preparing to unpack .../fluent-bit_2.0.6_amd64.deb ...
Unpacking fluent-bit (2.0.6) ...
Setting up libyaml-0-2:amd64 (0.2.5-1) ...
Setting up libpq5:amd64 (15.1-1) ...
Setting up fluent-bit (2.0.6) ...
Processing triggers for libc-bin (2.36-4) ...

Installation completed. Happy Logging!
```
